adapters/repos/db/shard.go

```diff
@@ -188,22 +184,24 @@ type onDeleteFromPropertyValueIndex func(shard *Shard, docID uint64, property *i
 type Shard struct {
 	index             *Index // a reference to the underlying index, which in turn contains schema information
 	class             *models.Class
-	queue             *VectorIndexQueue
-	queues            map[string]*VectorIndexQueue
 	scheduler         *queue.Scheduler
 	name              string
 	store             *lsmkv.Store
 	counter           *indexcounter.Counter
 	indexCheckpoints  *indexcheckpoint.Checkpoints
-	vectorIndex       VectorIndex
-	vectorIndexes     map[string]VectorIndex
 	metrics           *Metrics
 	promMetrics       *monitoring.PrometheusMetrics
 	slowQueryReporter helpers.SlowQueryReporter
 	propertyIndices   propertyspecific.Indices
 	propLenTracker    *inverted.JsonShardMetaData
 	versioner         *shardVersioner
 
+	// invariant: if vector index is present, then corresponding queue is also present and initialized
```

Comment:
suggestion: would it make sense to validate this invariant after shard initialization?

https://github.com/weaviate/weaviate/blob/dbb5cc8d7b21b11e7d3734dea662741eeeee13c0/adapters/repos/db/shard_init_vector.go#L44-L53

Reply:
Maybe, however, by that logic we should validate that all related variables are always initialized as well and that will become very verbose.

I guess I will simply remove this line to not make these fields in any way special. Code consumer will be able to deduce the invariant if it will inspect initialization logic.